src/beam/IFBeam.h

```diff
@@ -0,0 +1,34 @@
+#ifndef IFBEAM_H
+#define IFBEAM_H
+
+#include <string>
+#include <map>
+#include <vector>
+#include <nlohmann/json.hpp>
+
+#include "Params.h"
+#include "CAF.h"
+#include "reco/IRecoBranchFiller.h"
+
+using BeamSpills = std::map<double, double>;
+using json = nlohmann::json;
+
+class IFBeam {
+public:
+    IFBeam(const std::vector<std::vector<std::pair<const cafmaker::IRecoBranchFiller*, cafmaker::Trigger>>>& groupedTriggers, bool is_data);
+    
+    double getPOT(const cafmaker::Params& par, std::vector<std::pair<const cafmaker::IRecoBranchFiller*, cafmaker::Trigger>>& groupedTrigger, int ii);
+
+private:
+    const std::string potDevice = "E:TRTGTD";
+    BeamSpills beamSpills;
+
+    void loadBeamSpills(const std::vector<std::vector<std::pair<const cafmaker::IRecoBranchFiller*, cafmaker::Trigger>>>& groupedTriggers);
```

Comment:
While we're making aliases, adding 
```
using TriggerGroup = std::vector<std::pair<const cafmaker::IRecoBranchFiller*, cafmaker::Trigger>>;
```
to the `public` block of the class would simplify these a lot:
```suggestion
    IFBeam(const std::vector<std::vector<TriggerGroup>& groupedTriggers, bool is_data);
    
    double getPOT(const cafmaker::Params& par, TriggerGroup & groupedTrigger, int ii);

private:
    const std::string potDevice = "E:TRTGTD";
    BeamSpills beamSpills;

    void loadBeamSpills(const std::vector<TriggerGroup>& groupedTriggers);
```